dig_test.go

```diff
@@ -1905,6 +2024,46 @@ func TestProvideFailures(t *testing.T) {
 			`dig.out embeds \*dig.Out`,
 		)
 	})
+
```

Comment:
couple of additional tests:
 - use `Named` and `As` together (and verify the behaviour -- ensure both the function's return type and the `As` type are added to the container with the name).
 - provide a function that returns an interface, and then use `As`. If the `As` is for the same type, it should fail, but it should work if it's a different interface type